yarn-project/prover-node/src/prover-node.ts

```diff
@@ -115,32 +118,40 @@ export class ProverNode implements ClaimsMonitorHandler, EpochMonitorHandler {
    */
   async handleEpochCompleted(epochNumber: bigint): Promise<void> {
     try {
+      // Construct a quote for the epoch
       const blocks = await this.l2BlockSource.getBlocksForEpoch(epochNumber);
       const partialQuote = await this.quoteProvider.getQuote(Number(epochNumber), blocks);
       if (!partialQuote) {
         this.log.verbose(`No quote produced for epoch ${epochNumber}`);
         return;
       }
 
+      // Ensure we have deposited enough funds for sending this quote
+      await this.bondManager.ensureBond(partialQuote.bondAmount);
+
+      // Assemble and sign full quote
       const quote = EpochProofQuotePayload.from({
         ...partialQuote,
         epochToProve: BigInt(epochNumber),
         prover: this.publisher.getSenderAddress(),
         validUntilSlot: partialQuote.validUntilSlot ?? BigInt(Number.MAX_SAFE_INTEGER), // Should we constrain this?
```

Comment:
I'm not sure if there's an environment variable that does this but I think this should be constrained. As a default, it could be set to the first few slots of the epoch to be proven.